nat-lab/tests/conftest.py

```diff
@@ -67,6 +68,52 @@ def event_loop():
             loop.close()
 
 
+async def win_ports(vm_tag):
+    async def on_output(output: str) -> None:
+        print(f"win_ports_{vm_tag}: {output}")
+
+    start_port = random.randint(5000, 55000)
+    num_ports = random.randint(2000, 5000)
+    print(f"Setting up ports for {vm_tag}: start={start_port}, num={num_ports}")
+
+    async with new_connection_raw(vm_tag) as connection:
+        try:
+            await connection.create_process([
+                "netsh",
+                "int",
+                "ipv4",
+                "set",
+                "dynamic",
+                "tcp",
+                "start=" + str(start_port),
+                "num=" + str(num_ports),
+            ]).execute(on_output, on_output)
+        except ProcessExecError as e:
+            print(f"win_ports_{vm_tag} process execution failed: {e}")
+            return False
+
+    return True
+
+
+@pytest.fixture(autouse=True)
+@pytest.mark.windows
+def setup_windows_ports(request):
+    test_name = request.node.name
```

Comment:
I'd suggest using `request.session.items` then you could check for `marks`.

```
@pytest.fixture(autouse=True)
@pytest.mark.windows
def setup_windows_ports(request):
    items = request.session.items
    for item in items:
        for mark in item.own_markers:
            if mark.name != "windows":
                continue

            for tag in [ConnectionTag.WINDOWS_VM_1, ConnectionTag.WINDOWS_VM_2]:
                if not asyncio.run(win_ports(tag)):
                    print(f"Failed to setup ports for {tag}")

            return
```

Reply:
I incorporated your suggestion. So if there's a name defined test params - than only for that VM the command exectued. If there isn't - than for both.

Reply:
:+1: